core/src/main/java/com/datastax/oss/driver/internal/core/metadata/schema/DefaultTableMetadata.java

```diff
@@ -75,15 +75,23 @@ public CqlIdentifier getName() {
 
   @NonNull
   @Override
-  public UUID getId() {
-    return id;
+  public Optional<UUID> getId() {
+    if (id == null) {
```

Comment:
This can be simplified to `Optional.ofNullable(id)`.

Reply:
Also, you'll want to mark the `id` field as `Nullable`, currently this generates a warning.